app/controllers/teachers/sessions_controller.rb

```diff
@@ -1,33 +1,48 @@
 # frozen_string_literal: true
-
 class Teachers::SessionsController < Devise::SessionsController
   include TeacherCurrentNamespace
 
   layout "two_thirds"
 
-  def create
-    if resource_params[:create_or_sign_in] == "create"
-      redirect_to :eligibility_interface_countries
-      return
-    end
+  def new
+    @new_session_form = TeacherInterface::NewSessionForm.new
+  end
 
-    self.resource = resource_class.find_by(email: resource_params[:email])
+  def create
+    @new_session_form =
+      TeacherInterface::NewSessionForm.new(
+        email: new_session_form_params[:email],
+        create_or_sign_in: new_session_form_params[:create_or_sign_in]
+      )
 
-    if resource
-      if resource.active_for_authentication?
-        resource.send_magic_link
-      else
-        resource.resend_confirmation_instructions
+    if @new_session_form.valid?
+      if @new_session_form.create?
+        redirect_to :eligibility_interface_countries
+        return
       end
 
-      redirect_to :teacher_check_email
+      self.resource = resource_class.find_by(email: @new_session_form.email)
+
+      if resource
+        if resource.active_for_authentication?
+          resource.send_magic_link
+        else
+          resource.resend_confirmation_instructions
+        end
+        redirect_to :teacher_check_email
```

Comment:
Sorry, just one final spot, this line should go after the `end` below it, otherwise if we try and enter a valid email address that isn't in the database (i.e. `if resource` fails) we still want to take the user to the "check email" page as if it worked, to not expose the fact that the email address didn't exist.